uniffi_bindgen/src/templates/macros.rs

```diff
@@ -47,33 +52,63 @@
 {% macro to_rs_constructor_call(obj, cons) %}
 {% match cons.throws() %}
 {% when Some with (e) %}
-UNIFFI_HANDLE_MAP_{{ obj.name()|upper }}.insert_with_result(err, || -> Result<{{obj.name()}}, {{e}}> {
-    let _retval = {{ obj.name() }}::{% call to_rs_call(cons) %}?;
-    Ok(_retval)
-})
+    todo!("XXX - catch_unwind like below...");
+    let constructed = {{ obj.name() }}::{% call to_rs_call(cons) %}?;
+    let arc = std::sync::Arc::new(constructed);
+    Ok(std::sync:Arc::into_raw(arc))
 {% else %}
-UNIFFI_HANDLE_MAP_{{ obj.name()|upper }}.insert_with_output(err, || {
-    {{ obj.name() }}::{% call to_rs_call(cons) %}
-})
+    match std::panic::catch_unwind(|| {
+        {%- if obj.threadsafe() %}
+        let _new = {{ obj.name() }}::{% call to_rs_call(cons) %};
+        {%- else %}
+        let _new = std::sync::Mutex::new({{ obj.name() }}::{% call to_rs_call(cons) %});
+        {%- endif %}
+        let _arc = std::sync::Arc::new(_new);
+        std::sync::Arc::into_raw(_arc)
```

Comment:
(I know this isn't intended for merge, but I'm commenting to share my mental model of how the final code might look).

This needs a comment along the lines of "This copy of the Arc is leaked, and is conceptually now owned by the foreign-language code. It will only be dropped when the foreign-language code calls the matching desctructor".